addons/web/static/src/views/fields/input_field_hook.js

```diff
@@ -18,29 +18,62 @@ export function useInputField(params) {
     const env = useEnv();
     const inputRef = useRef(params.refName || "input");
     const component = useComponent();
+
+    /*
+     * An field is dirty if it is no longer sync with the model
+     * More specifically, a field is no longer dirty after it has *tried* to update the value in the model.
+     * An invalid value will thefore not be dirty even if the model will not actually store the invalid value.
+     */
     let isDirty = false;
+
+    /**
+     * A field is invalid if the parsing of its value failed.
+     */
+    let isInvalid = false;
+
+    /**
+     * The last value that has been comited to the model.
+     * Not changed in case of invalid field value.
+     */
     let lastSetValue = null;
+
+    /**
+     * When a user types, we need to set the field as dirty.
+     */
     function onInput(ev) {
         isDirty = ev.target.value !== lastSetValue;
         if (component.props.setDirty) {
             component.props.setDirty(isDirty);
         }
     }
+
+    /**
+     * On blur, we consider the field no longer dirty, even if it were to be invalid.
+     * However, if the field is invalid, the new value will not be committed to the model.
+     */
     function onChange(ev) {
-        lastSetValue = ev.target.value;
         isDirty = false;
```

Comment:
shouldn't we still check here if isDirty is false, and discard the all thing if it is ?

Reply:
perhaps also the invalide state, not sure about that it's touchy. 
But it is checked in commitChanges for now.

Reply:
nope